frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/divider/divider_character_shortcut_event.dart

```diff
@@ -18,27 +18,48 @@ final CharacterShortcutEvent convertMinusesToDivider = CharacterShortcutEvent(
 CharacterShortcutEventHandler _convertMinusesToDividerHandler =
     (editorState) async {
   final selection = editorState.selection;
+  
   if (selection == null || !selection.isCollapsed) {
     return false;
   }
+
   final path = selection.end.path;
   final node = editorState.getNodeAtPath(path);
   final delta = node?.delta;
+
   if (node == null || delta == null) {
     return false;
   }
-  if (delta.toPlainText() != '--') {
+
+  if (!_hasTwoConsecutiveDashes(node.delta!.toPlainText(), selection.start.offset)) {
     return false;
   }
-  final transaction = editorState.transaction
-    ..insertNode(path, dividerNode())
-    ..insertNode(path, paragraphNode())
-    ..deleteNode(node)
-    ..afterSelection = Selection.collapse(path.next, 0);
+
+  final dashStartPosition = selection.start.offset - 2;
```

Comment:
I'd avoid all uses of `2` to calculate. It looks like we're using it here because `--` is two characters long. If we decide to change the shortcut characters to, say, for example, `---` then we will have to update all uses of this calculation.

```suggestion
  final dashStartPosition = selection.start.offset - dividerShortcutToken.length;
```